tests/bluetooth/host/buf/bt_buf_get_cmd_complete/src/main.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <ztest.h>
+#include "kconfig.h"
+#include "hooks.h"
+#include "buf_help_utils.h"
+
+#include <zephyr/bluetooth/hci.h>
+#include <host/hci_core.h>
+#include <zephyr/bluetooth/buf.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/slist.h>
+#include <zephyr/sys/util.h>
+
+/*
+ *  Return value from bt_buf_get_cmd_complete() should be NULL
+ *
+ *  This is to test the behaviour when memory allocation request fails
+ *
+ *  Constrains:
+ *     - bt_dev.sent_cmd value is NULL
+ *     - Timeout value is a positive non-zero value
+ *     - net_buf_alloc_fixed() returns a NULL value
+ *
+ *  Expected behaviour:
+ *     - net_buf_alloc_fixed() to be called with the correct memory allocation pool
+ *       and the same timeout value passed to bt_buf_get_cmd_complete()
+ *     - bt_buf_get_cmd_complete() returns NULL
+ *
+ */
+void test_returns_null_sent_cmd_is_null(void)
+{
+	struct net_buf *buf;
+	k_timeout_t timeout = Z_TIMEOUT_TICKS(1000);
+
+	bt_dev.sent_cmd = NULL;
+
+#if defined(CONFIG_BT_HCI_ACL_FLOW_CONTROL)
+	ztest_expect_value(net_buf_alloc_fixed, pool, bt_buf_get_evt_pool());
+#else
+	ztest_expect_value(net_buf_alloc_fixed, pool, bt_buf_get_hci_rx_pool());
+#endif
+
+	ztest_returns_value(net_buf_alloc_fixed, NULL);
+	ztest_expect_value(hooks_net_buf_alloc_fixed_timeout_validation_hook, value, timeout.ticks);
```

Comment:
@ahmedmoheb-nordic see e.g. #48074 for an example.

Reply:
If you don't want to use FFF, I'd suggest renaming that fn to `net_buf_alloc_timeout_value`, since the function is just a mock to extract the tick value and you do the mock setup in the `main.c`.